### DISCUSSION TITLE
Help with three things, memory use estimation, circular slabs for inverse caustics and plotting out of focus simulations

### DISCUSSION THREAD

**Discussion by tstigen** [mitsuba-renderer/mitsuba3#1037] [Q&A]
Hello, 

I finally got mitsuba working after some driver reversions. I'm interested in the inverse caustics and can get the tutorial to work, but there are some modifications I'm trying and not having any luck with.

A) The tutorial uses a square slab for the lens. I would like to have a "regular" circular lens. But I'm unsure how to go about this as the square slab is loading a .obj file. Do I have to make a cylindrical .obj and define new logic for a radial meshgrid? How does that work with the triangular mesh tessellation in the optimization step? 

B) Once I have a successful inverse caustic heightmap defined, I want to plot it under difference scenarios to see what "focusing" the lens looks like. I attempted to do this by making a new scene with the same parameters as the old scene but changing the distance of the receiver. I could never get the new scene to render anything, even if I didn't change the focus. 

C) For the inverse caustics example, how would one estimate the GPU memory being used? Especially with the core parameters
  'render_resolution': (128, 128),
        'heightmap_resolution': (512, 512),
        'n_upsampling_steps': 4,
        'spp': 16,
        'max_iterations': 1000,

I find that if I increase heightmap to 1024x1024, I surpass my 8GB somewhere around 900ish iterations and it crashes. I'm wondering if there is a way to piecemeal it if I know how much memory a certain parameter layout would produce. Then I could run up to that limit, save the state and clear memory and reinitialize and continue some more. 

Attached my python notebook. The last cell is where I'm trying to plot the unfocused lens, but just get static. 

[runme.zip](https://github.com/mitsuba-renderer/mitsuba3/files/13926786/runme.zip)

**Comment by wjakob** [comment]
Hi @tstigen -- our these questions are unfortunately beyond the scope of what we can support here. The out of memory issue is likely a bug that will be fixed in an upcoming version of Mitsuba.
